### ISSUE TITLE
Invalid container interop call

### ISSUE THREAD

**Issue by wdalmut** [pennyphp/penny#77]
In the `__construct you use a`set`method on the container but effectively the`ContainerInterface` doesn't implement this method...

https://github.com/container-interop/container-interop/blob/master/src/Interop/Container/ContainerInterface.php

here where you use the `set`

https://github.com/pennyphp/penny/blob/master/src/App.php#L53

You can only use:
- `get`
- `has`

That's right?

**Comment by gianarb** [comment]
:bomb: right!

We have a little problem of consistency

I'm thinking different solution..
1) https://github.com/jeremeamia/acclimate-container

2) DiFactories
This method is in the wrong place
https://github.com/pennyphp/penny/blob/master/src/App.php#L65L77

We can move it into the Penny\Container\PHPDiFactory I use it if you don't specify nothing into the App(**$container**).. 
and you can use your implementation of this factory.. Penny\Container\ServiceLocatorFactory

``` php
    public static function buildContainer($config = [])
    {
        $container = new ServiceManager();

        $container->setFactory('event_manager', function ($container) {
            return new Zend\EventManager\EventManager();
        });
        // configure router and dispatcher

       return $container
    }
```

**Comment by wdalmut** [comment]
;)

**Comment by wdalmut** [comment]
Why the `__constructor` method should not limited to:

``` php
    public function __construct($router = null, ContainerInterface $container = null)
    {
        $this->container = $container ?: static::buildContainer(Loader::load());
        $container = &$this->container;

        $this->response = new Response();
        $this->request = ServerRequestFactory::fromGlobals();

        if ($router == null && $container->has('router') == false) {
            throw new Exception('Define router config');
        }
    }
```

Why you want to set the DiC in the DiC definitions? and check if the router exists after an exception throws?

``` php
// Constructor extralines at the bottom of `__constructor`
-        if ($container->has('router') == false) {
-            $container->set('router', $router);
-        }
-
-        $container->set('di', $container);
```

**Comment by gianarb** [comment]
I set DiC into the DiC to resolve this use case
https://github.com/pennyphp/penny/blob/master/tests/app/Controller/Index.php#L14

**Comment by gianarb** [comment]
Resolved #79
